cpp/cmake/thirdparty/get_raft.cmake

```diff
@@ -45,14 +45,25 @@ function(find_and_configure_raft)
       COMPONENTS ${RAFT_COMPONENTS}
         CPM_ARGS
             EXCLUDE_FROM_ALL TRUE
-            GIT_REPOSITORY https://github.com/${PKG_FORK}/raft.git
-            GIT_TAG        ${PKG_PINNED_TAG}
+            GIT_REPOSITORY https://github.com/jameslamb/raft.git
+            GIT_TAG        use-filesets
             SOURCE_SUBDIR  cpp
             OPTIONS
                 "RAFT_COMPILE_LIBRARY ${PKG_COMPILE_RAFT_LIB}"
                 "BUILD_TESTS OFF"
                 "BUILD_BENCH OFF"
                 "BUILD_CAGRA_HNSWLIB OFF"
+                "BUILD_PRIMS_BENCH OFF"
+    )
+
+    # install the RAFT headers
+    # TODO(jameslamb): get exporting working
+    # TODO(jameslamb): move this under if(raft_ADDED)?
+    install(
+      TARGETS raft
+      EXPORT cugraph-exports
+      FILE_SET raft_headers
+      DESTINATION ${CMAKE_INSTALL_PREFIX}
     )
```

Comment:
Alright, this is getting closer! But I need some CMake help. Here's what I'm trying to do in `libcugraph` wheels:

* build static RAFT (`libraft.a`) and statically link it into `libcugraph.so`
* do not install `libraft.a` in `libcugraph` wheels
* do not built `libraft.so` at all (or at least, don't install it into `libcugraph` wheels)
* install the RAFT headers into the `libcugraph` wheels, and export them
* use `libcugraph` wheels in the build env for `pylibcugraph` and `cugraph` wheels, and have `libcugraph`'s PUBLIC `raft::raft` dependency be served by the RAFT headers it vendors

I've got all the RAFT headers installing into `include/raft/` in the wheel the way I want, thanks to the combination of:

* setting `EXCLUDE_FROM_ALL TRUE` in the `rapids_cpm_find(raft)` call above
* this `install(TARGETS`
* using `target_sources()` + `FILE_SET` to associated the header files with the `raft` target in RAFT (https://github.com/rapidsai/raft/pull/2528)

But I can't figure out the exporting piece.

The current state of this branch results in stuff like this at configure time for `libcugraph`:

>   CMake Error: install(EXPORT "raft-compiled-exports" ...) includes target "raft_compiled" which requires target "raft" that is not in this export set, but in multiple other export sets: lib64/cmake/cugraph/cugraph-targets.cmake, lib64/cmake/raft/raft-targets.cmake.
  An exported target cannot depend upon another target which is exported multiple times. Consider consolidating the exports of the "raft" target to a single export.

>  CMake Error: install(EXPORT "raft-compiled-exports" ...) includes target "raft_compiled_static" which requires target "raft" that is not in this export set, but in multiple other export sets: lib64/cmake/cugraph/cugraph-targets.cmake, lib64/cmake/raft/raft-targets.cmake.

>   CMake Error: install(EXPORT "raft-compiled-lib-exports" ...) includes target "raft_lib" which requires target "raft" that is not in this export set, but in multiple other export sets: lib64/cmake/cugraph/cugraph-targets.cmake, lib64/cmake/raft/raft-targets.cmake.

If I remove the `EXPORT` line, `libcugraph` builds successfully but then the `pylibcugraph` builds cannot find those RAFT headers (because no `raft-config.cmake` is written out):

> -- Configuring done (3.9s)
CMake Error at /pyenv/versions/3.12.7/lib/python3.12/site-packages/libcugraph/lib64/cmake/cugraph/cugraph-targets.cmake:61 (set_target_properties):
  The link interface of target "cugraph::cugraph" contains:
>    raft::raft
>  but the target was not found.  Possible reasons include:
>    * There is a typo in the target name.
>    * A find_package call is missing for an IMPORTED target.
>    * An ALIAS target is missing.
> Call Stack (most recent call first):
>  /pyenv/versions/3.12.7/lib/python3.12/site-packages/libcugraph/lib64/cmake/cugraph/cugraph-config.cmake:73 (include)
>  CMakeLists.txt:30 (find_package)

If I remove this `install()` and remove `EXCLUDE_FROM_ALL TRUE` from the `rapids_cpm_find(raft)` call, then the RAFT headers, `libraft.a`, and `libraft.so` are all packaged in the `libcugraph` wheel and exported successfully, such that `pylibcugraph` and `cugraph` wheel builds find them and succeed ([without runtime issues, at least in the one test job I ran](https://github.com/rapidsai/cugraph/actions/runs/12241843897/job/34149484561)).

I think I'm just fundamentally misunderstanding some things about dependency tracking in CMake / `rapids-cmake`. I've tried various forms of:

* `rapids_export()`
* `rapids_export_package()`
* `install(EXPORT`

And haven't yet found the right combination.

If what I'm trying to do is not possible or just undesirable for something other reason, then other approaches to try (some of which are from an offline discussion with @bdice and @vyasr) include:

* omit `EXCLUDE_FROM_ALL TRUE` and just add a hack to the `libcugraph` CMakeLists.txt that manually deletes `libraft.a` and `libraft.so`
* have `pyblibcugraph` and `cugraph` wheel builds re-download all the RAFT headers via CPM at build time
* stop on this and first get `libraft` wheels set up, then use them at build time for `libcugraph`, `pylibcugraph`, and `cugraph`

Reply:
I've started exploring whether we can avoid needing to worry about this by just creating `libraft` wheels: https://github.com/rapidsai/raft/pull/2531

We can come back to this thread if that doesn't work out.